helper/intersectionsParsing.js

```diff
@@ -0,0 +1,134 @@
+const _ = require('lodash');
+/*
+this function returns an object that denotes an intersection of form:
+{
+  street1: value1,
+  street2: value2
+}
+*/
+module.exports = function parseIntersections(text) {
+    var str1 = '', str2 = '';
+    if(text.trim().length > 1) {
+        //   var words = text.toLowerCase().split(' ');
+        var words = _.words(text.toLowerCase(), /[^ ]+/g);
+
+        // changes 'e15' to 'East 15', etc.
+        words = directionalSanitizer(words);
+        // changes '6' to '6th', etc
+        words = addOrdinality(words);
+        // only treat input as intersection if contains '&' or 'and'
+        const delimiter = _.includes(text, '&') ? '&' : 'and';
+        const delimiterIndex = words.indexOf(delimiter);
+
+        str1 = wordsToSentence(words, 0, delimiterIndex);
+        str2 = wordsToSentence(words, delimiterIndex+1, words.length);
+    } else {
+        throw 'Missing streets in the intersection';
+    }
+    return { street1: str1, street2: str2 };
+};
+
+// intended for intersections only
+// this function turns '77' into '77th', '3' into '3rd', etc
+function addOrdinality(arr) {
+    arr.forEach( function (elmnt, index) {
+        // is it only numbers
+        let isNum = /^\d+$/.test(elmnt);
+        if(isNum) {
+            switch(elmnt[elmnt.length-1]){
+                case '1':
+                    elmnt += 'st';
+                    arr[index] = elmnt;
+                    break;
+                case '2':
+                    elmnt += 'nd';
+                    arr[index] = elmnt;
+                    break;
+                case '3':
+                    elmnt += 'rd';
+                    arr[index] = elmnt;
+                    break;
+                default :
+                    elmnt += 'th';
+                    arr[index] = elmnt;
+            }
+        }
+    });
+    return arr;
+}
+
+// intended to do the conversions like:
+// 'w28' -> 'West 28'
+// 'e17' -> 'East 17'
+
+function directionalSanitizer(arr){
+    const mapping = {
+        e : 'East',
+        w : 'West',
+        s : 'South',
+        n : 'North',
+        ne: 'Northeast',
+        nw: 'Northwest',
+        se: 'Southeast',
+        sw: 'Southwest'
+    };
+
+    for (let i = 0; i < arr.length; i++) {
```

Comment:
This is another case where `Array.map` would really help. instead of modifying the incoming array, returning a new array with appropriate changes made is generally easier to read and make correct.

Reply:
@orangejulius Done!